components/devtools/lib.rs

```diff
@@ -154,7 +154,14 @@ fn run_server(
     port: u16,
     embedder: EmbedderProxy,
 ) {
-    let listener = TcpListener::bind(&("0.0.0.0", port)).unwrap();
+    let listener = match TcpListener::bind(&("0.0.0.0", port)) {
+        Ok(listener) => listener,
+        _ => TcpListener::bind(&("0.0.0.0", 0u16)).unwrap(),
+    };
```

Comment:
The port number comes from a preference. We should keep it that way. Falling back to `0` on fail might create unexpected behavior (what if it failed for another reason than a busy port?). Also, do not `unwrap`, it will crash Servo if the devtools can't start.

My recommendation:
- revert that change: https://github.com/servo/servo/pull/25899
- instead of `6000`, use `0`
- The message `OnDevtoolsStarted` could be: `OnDevtoolsStarted(Result<u16>)`, then if it fails, it would notify the embedder

Reply:
What type of Error should be returned? I tried using the `std::io::Result<T>` (returned by `bind()` and `local_addr()`, but since `std::io::Error` doesn't implement Serialize trait, it cannot be used.

Reply:
Just `()` will be fine.